storage/imx6ull-flash/flashdev.c

```diff
@@ -99,113 +100,130 @@ static unsigned int _flashmtd_checkErased(const void *buff, size_t boffs, size_t
-		switch (meta->errors[i]) {
-			case flash_no_errors:
-			case flash_erased:
-				break;
+	switch (meta->errors[chunkidx]) {
+		case flash_no_errors:
+		case flash_erased:
+			ret = 0;
+			break;
 
-			case flash_uncorrectable:
-				/* BCH reports chunk as uncorrectable in case of an erased page with bitflips within that chunk area */
-				/* Check if that's the case by counting the bitflips with an assumption that the page is fully erased */
-
-				/* Read raw page in order to check full chunk - both data and its ECC area */
-				if (raw == NULL) {
-					/* Map raw page buffer */
-					rawsz = (strg->dev->mtd->writesz + strg->dev->mtd->metaSize + _PAGE_SIZE - 1) & ~(_PAGE_SIZE - 1);
-					raw = mmap(NULL, rawsz, PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_CONTIGUOUS, 0);
-					if (raw == MAP_FAILED) {
-						return -ENOMEM;
-					}
-
-					/* Read raw page */
-					err = flashdrv_readraw(strg->dev->ctx->dma, paddr, raw, strg->dev->mtd->writesz + strg->dev->mtd->metaSize);
-					if (err < 0) {
-						munmap(raw, rawsz);
-						return err;
-					}
-				}
-
-				/* Metadata chunk */
-				if (i == 0) {
-					boffs = 0;
-					blen = mlen;
-				}
-				/* Data chunk */
-				else {
-					boffs = mlen + (i - 1) * dlen;
-					blen = dlen;
-				}
-
-				flips = _flashmtd_checkErased(raw, boffs, blen);
-
-				/* Nothing to do if there're no bitflips */
-				if (flips == 0) {
-					break;
-				}
-
-				if (i == 0) {
-					/* Correct metadata chunk */
-					memset(meta, 0xff, strg->dev->mtd->oobSize);
-				}
-				else {
-					/* Correct data chunk */
-					memset(data + (i - 1) * ECCN_DATA_SIZE, 0xff, ECCN_DATA_SIZE);
-				}
-
-				/* Chunk corrected, update max number of bitflips */
-				maxflips = max(flips, maxflips);
+		case flash_uncorrectable:
+			/* BCH reports chunk as uncorrectable in case of an erased page with bitflips within that chunk area */
+			/* Check if that's the case by counting the bitflips with an assumption that the page is fully erased */
+			/* Read raw page in order to check full chunk - both data and its ECC area */
+			raw = mmap(NULL, rawsz, PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_CONTIGUOUS, 0);
+			if (raw == MAP_FAILED) {
+				ret = -ENOMEM;
 				break;
+			}
 
-			default:
-				/* Chunk corrected by BCH, update max number of bitflips */
-				maxflips = max(meta->errors[i], maxflips);
+			/* Read raw page */
+			ret = flashdrv_readraw(ctx->dma, pageaddr, raw, mtd->writesz + mtd->metaSize);
+			if (ret < 0) {
 				break;
-		}
+			}
+
+			if (CHUNK_IS_META(chunkidx)) {
+				boffs = 0;
+				blen = metalen;
+			}
+			else {
+				boffs = metalen + (chunkidx - 1) * datalen;
+				blen = datalen;
+			}
+
+			ret = _flashmtd_checkErased(raw, boffs, blen);
+
+			/* Let the user see this chunk as correctly erased */
+			_flashmtd_chunkCorrect(strg, chunkidx);
+			break;
+
+		default:
+			/* On any other case metadata error field contains number of corrected bits */
+			ret = meta->errors[chunkidx];
 	}
 
 	if (raw != NULL) {
```

Comment:
&& raw != MAP_FAILED

Reply:
not applied

Reply:
I have applied it. I set raw to NULL in case mmap returns MAP_FAILED:
https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/4069c06d0c116e73117022a361fd4ae3db5b0550/storage/imx6ull-flash/flashdev.c#L153

Reply:
sorry, my bad, not used to GH PRs and how to clearly see diffs between patchsets